persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/impl/table/AbstractTableManager.java

```diff
@@ -162,7 +162,7 @@ public void createMetaTable(Connection conn) throws PersistenceException {
    @Override
    public void updateMetaTable(Connection conn) throws PersistenceException {
       short version = Version.getVersionShort();
-      int segments = ctx.getCache().getCacheConfiguration().clustering().hash().numSegments();
+      int segments = ctx.getConfiguration().segmented() ? ctx.getCache().getCacheConfiguration().clustering().hash().numSegments() : -1;
```

Comment:
Seems this JDBC stuff should be a different PR?

Reply:
It was uncovered by the changes to `PersistenceCompatibilityTest`. I think it needs to be part of this PR as this could cause issues in the operator upgrade case in a similar way to the SFS issues.